lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/LocalStorageClient.java

```diff
@@ -197,4 +96,86 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOEx
       return FileVisitResult.CONTINUE;
     }
   }
+
+
+  private class LocalFileReference extends BaseFileReference {
+
+    private final Path path;
+
+    // information that other FileReferences usually have access to via the storage object.
+    private final Instant creationTime;
+    private final long size;
+
+    // The given path will become absolute and will be normalized.
+    public LocalFileReference(Path path, BasicFileAttributes attributes) {
+      super(attributes.lastModifiedTime().toInstant());
+
+      this.path = path.toAbsolutePath().normalize();
+      this.creationTime = attributes.creationTime().toInstant();
+      this.size = attributes.size();
+    }
+
+    @Override
+    public String getName() { return path.getFileName().toString(); }
+
+    @Override
+    public String getFullPath(TraversalParams params) {
+      return path.toString();
+    }
+
+    @Override
+    public boolean isValidFile() {
+      return true;
+    }
+
+    @Override
+    public InputStream getContentStream(TraversalParams params) {
```

Comment:
should add javadoc saying that the returned InputStream is non-buffered
depending on how the caller is going to use this stream, the caller might want to wrap it in a BufferedInputStream